sdk/identity/src/oauth2_http_client.rs

```diff
@@ -0,0 +1,110 @@
+//! Implements the oauth2 crate http client interface using an `azure_core::HttpClient` instance.
+//! https://docs.rs/oauth2/latest/oauth2/#importing-oauth2-selecting-an-http-client-interface
+
+use azure_core::{
+    error::{Error, ErrorKind, ResultExt},
+    HttpClient, Request,
+};
+use std::{collections::HashMap, str::FromStr, sync::Arc};
+
+pub(crate) struct Oauth2HttpClient {
+    http_client: Arc<dyn HttpClient>,
+}
+
+impl Oauth2HttpClient {
+    /// Create a new Oauth2HttpClient
+    pub fn new(http_client: Arc<dyn HttpClient>) -> Self {
+        Self { http_client }
+    }
```

Comment:
I thought we were moving away from new_http_client and moving to `Pipeline`?

Reply:
I guess the question is whether identity should also go through a `Pipeline`. It makes sense in a way. The challenge is that there is specific logic the oauth client handles for us that a simple transport policy would not do.

Reply:
Created #1015 for this.